Give the ORM docs a read through and fix old content (#7808)
cakephp/docs: en/orm/retrieving-data-and-resultsets.rst

ways to inspect the data returned by the ORM.

- ``debug($query)`` Shows the SQL and bound parameters, does not show results.

> - ``sql($query)`` Shows the final rendered SQL, but only when you have DebugKit installed.

Reviewer: ```suggestion
- ``sql($query)`` Shows the final rendered SQL when DebugKit is installed.
```
